File: NYUDining/RFHoursTableViewController.m
//
//  HoursTableViewController.m
//  NYUDining
//
//  Created by Ross Freeman on 11/5/15.
//  Copyright © 2015 Ross Freeman. All rights reserved.
//

#import "RFHoursTableViewController.h"

@interface RFHoursTableViewController ()

@end

@implementation RFHoursTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [_hoursTable reloadData];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table View

-(NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 7;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"hours"];
    
    int index = (int)indexPath.row;
    NSString *hours = _diningLocation.hours[index];
    NSLog(@"%@", hours);
    
    NSString *dayOfWeek = @"";
    
    switch (index) {
        case 0:
            dayOfWeek = @"Sunday";
            break;
        case 1:
           dayOfWeek = @"Monday";
            break;
        case 2:
            dayOfWeek = @"Tuesday";
            break;
        case 3:
            dayOfWeek = @"Wednesday";
            break;
        case 4:
            dayOfWeek = @"Thursday";
            break;
        case 5:
            dayOfWeek = @"Friday";
            break;
        default:
            dayOfWeek = @"Saturday";
            break;
    }
    
    cell.textLabel.text = dayOfWeek;
    cell.detailTextLabel.text = hours;
    
    return cell;
}

@end

File: NYUDining/RFHoursTableViewController.swift
//
//  RFHoursTableViewController.swift
//  NYUDining
//
//  Created by Ross Freeman on 6/10/16.
//  Copyright © 2016 Ross Freeman. All rights reserved.
//

import UIKit
import PureLayout

class RFHoursTableViewController: UIViewController {
    
    var diningLocation: RFDiningLocation!
    fileprivate let hoursTable = UITableView()
    
    fileprivate let cellId = "hours"
    fileprivate var didSetConstraints = false

    override func viewDidLoad() {
        super.viewDidLoad()
        
        hoursTable.register(HoursTableViewCell.self, forCellReuseIdentifier: cellId)
        hoursTable.dataSource = self
        hoursTable.isUserInteractionEnabled = false
        
        view.addSubview(hoursTable)

        view.setNeedsUpdateConstraints()
        hoursTable.reloadData()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func updateViewConstraints() {
        if !didSetConstraints {
            hoursTable.autoPinEdgesToSuperviewEdges()
        }
        
        super.updateViewConstraints()
    }

}

extension RFHoursTableViewController: UITableViewDataSource {
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        if let hours = diningLocation.hours {
            return hours.count
        }
        
        return 0
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: cellId, for: indexPath) as! HoursTableViewCell
        
        var hours = diningLocation.hours[(indexPath as NSIndexPath).row]
        
        var dayOfWeek: DayOfWeek
        
        switch (indexPath as NSIndexPath).row {
        case 0:
            dayOfWeek = .Sunday
        case 1:
            dayOfWeek = .Monday
        case 2:
            dayOfWeek = .Tuesday
        case 3:
            dayOfWeek = .Wednesday
        case 4:
            dayOfWeek = .Thursday
        case 5:
            dayOfWeek = .Friday
        default:
            dayOfWeek = .Saturday
        }
        
        cell.weekdayLabel.text = dayOfWeek.rawValue
        
        hours = hours.replacingOccurrences(of: ",", with: "\n")
        cell.hoursLabel.text = hours
        
        
        return cell
        
    }
}
